File: minegame-common/src/main/java/top/nipuru/minegame/common/message/shared/ListingMessage.java
package top.nipuru.minegame.common.message.shared;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;

@Getter
@Setter
@Accessors(chain = true)
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ListingMessage implements Serializable {
    /** 主键id */
    long listingId;

    /** 出售者uuid */
    int sellerId;

    /** 购买者uuid */
    int buyerId;

    /** 交易时间 */
    long dealTime;

    /** 过期时间 */
    long expireTime;

    /** 分组 */
    int category;

    /** 价格 */
    int price;

    /** 物品名称 */
    String name;

    /** 物品数据 */
    byte[] item;
}

File: minegame-common/src/main/kotlin/top/nipuru/minegame/common/message/shared/ListingMessage.kt
package top.nipuru.minegame.common.message.shared

import java.io.Serializable

class ListingMessage : Serializable {
    /** 主键id  */
    var listingId: Long = 0

    /** 出售者uuid  */
    var sellerId: Int = 0

    /** 购买者uuid  */
    var buyerId: Int = 0

    /** 交易时间  */
    var dealTime: Long = 0

    /** 过期时间  */
    var expireTime: Long = 0

    /** 分组  */
    var category: Int = 0

    /** 价格  */
    var price: Int = 0

    /** 物品名称  */
    var name: String = ""

    /** 物品数据  */
    var item: ByteArray = ByteArray(0)
}
